### PR TITLE
Aavruti_Assignment_35

### PR DESCRIPTION
**Pull request by AavrutiDJun20**
Union of two Arrays Using Collection

**After Modification:**
image

**BeforeModification:**
image

### CODE REVIEW

File: src/aavrutiDJun20/collectionPrgm/UnionOfArrayUsingList.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	}*/
+
+	//Union of two Arrays
+	void findUnionOfArray(ArrayList<String> nameList1, ArrayList<String> nameList2) {
+		ArrayList<String> list1 = new ArrayList<String>(nameList1);
+		list1.addAll(nameList2);
+
+		for(int index=0;index<list1.size();index++) {
+			if(list1.indexOf(list1.get(index)) != list1.lastIndexOf(list1.get(index))) {
+				list1.remove(list1.get(index));
```
Comment (AditiGJUN20) [inline src/aavrutiDJun20/collectionPrgm/UnionOfArrayUsingList.java]: **Suggestion:**
You can use `list1.remove(list1.lastIndexOf(list1.get(index)))` to match the results to expected output.

File: src/aavrutiDJun20/collectionPrgm/UnionOfArrayUsingList.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		System.out.println(list1);
+	}
+
+	public static void main(String[] args) {
+		UnionOfArrayUsingList unionOfArrayUsingList = new UnionOfArrayUsingList();
+		String[] name1 = {"Palak", "Viresh", "Yash", "Aavruti", "Viresh"};
+		String[] name2 = {"Deavina","Palak","Viresh", "Nikhil", "Nikhil"};		
+		ArrayList<String> nameList1 = new ArrayList<String>(Arrays.asList(name1));
+		ArrayList<String> nameList2 = new ArrayList<String>(Arrays.asList(name2));		
+		unionOfArrayUsingList.findUnionOfArray(nameList1,nameList2);
```
Comment (AditiGJUN20) [inline src/aavrutiDJun20/collectionPrgm/UnionOfArrayUsingList.java:48]: Code is not working for following input:
`String[] name1 = { "Palak", "Viresh", "Yash", "Aavruti","Palak","Yash", "Yash" }; 
String[] name2 = { "Deavina","Palak", "Nikhil", "Palak", "Viresh", "Nikhil", "Nikhil","Yash" };	`
With current code following output is coming:
`Unique List: [Aavruti, Yash, Deavina, Palak, Viresh, Nikhil, Nikhil, Yash] `

**Review by AditiGJUN20 — CHANGES_REQUESTED** [review]
Code working for given input. Not working when input array has multiple occurrence of same element.

File: src/aavrutiDJun20/collectionPrgm/UnionOfArrayUsingList.java (hunk 1 above)
Comment (AavrutiDJun20) [reply src/aavrutiDJun20/collectionPrgm/UnionOfArrayUsingList.java]: Thank You
Made changes as per your suggestion.

File: src/aavrutiDJun20/collectionPrgm/UnionOfArrayUsingList.java (hunk 2 above)
Comment (AavrutiDJun20) [reply src/aavrutiDJun20/collectionPrgm/UnionOfArrayUsingList.java:48]: Made changes as per your suggestion.
Thank you for taking so much effort and checking this additional scenario and letting me know.
I will keep it in my mind for next time.

File: src/aavrutiDJun20/collectionPrgm/UnionOfArrayUsingList.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+	void findUnionOfArray(ArrayList<String> nameList1, ArrayList<String> nameList2) {
+		ArrayList<String> list1 = new ArrayList<String>(nameList1);		
+		list1.addAll(nameList2);		
+		
+		for(int index=0;index<list1.size();) {
+			String name = list1.get(index);
+			if(list1.indexOf(name) != list1.lastIndexOf(name)) {				
+				list1.remove(list1.lastIndexOf(name));
+				continue;
```
Comment (AditiGJUN20) [inline src/aavrutiDJun20/collectionPrgm/UnionOfArrayUsingList.java]: **Suggestion:**
Code will work even `continue` is not written.

**Review by AditiGJUN20 — APPROVED** [review]
Good Job Aavruti. 
Code looks good to me. Approved for further Review.

**Review by KrishnaKTechnocredits — APPROVED** [review]
Code looks good to me.

**Pull request merged**